src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -349,7 +349,7 @@ class ProtobufPlugin implements Plugin<Project> {
         project.plugins.withId("org.jetbrains.kotlin.android") {
           project.afterEvaluate {
             String compileKotlinTaskName = Utils.getKotlinAndroidCompileTaskName(project, variant.name)
-            project.tasks.named(compileKotlinTaskName, SourceTask) { SourceTask task ->
+            project.tasks.named(compileKotlinTaskName, KotlinCompile) { KotlinCompile task ->
```

Comment:
So #637 was the source of the regression, it appears because this had previously been dynamically typed. Makes sense how that happens.

Reply:
KotlinCompile was SourseTask, but kotlin team removed that extends.

Reply:
So they broke API in a minor release? Ugh. The cause for the regression is still because this is no longer dynamic, just we weren't testing the right version to notice.

Reply:
Kotlin [v1.6.20](https://github.com/JetBrains/kotlin/blob/v1.6.20/libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/tasks/Tasks.kt#L78) AbstractKotlinCompileTool extends AbstractCompile, AbstractCompile extends SourceTask.

Kotlin [v1.7.20](https://github.com/JetBrains/kotlin/blob/v1.7.20/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/tasks/Tasks.kt#L77) AbstractKotlinCompileTool extends DefaultTask.

¯\_(ツ)_/¯